src/public/services/EncryptionService.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  AttachmentResponse,
+  CheckboxResponse,
+  DateResponse,
+  DecimalResponse,
+  DropdownResponse,
+  EmailResponse,
+  FieldResponse,
+  HeaderResponse,
+  HomeNoResponse,
+  LongTextResponse,
+  MobileResponse,
+  NricResponse,
+  NumberResponse,
+  RadioResponse,
+  RatingResponse,
+  ShortTextResponse,
+  TableResponse,
+  UenResponse,
+  YesNoResponse,
+} from '../../../shared/types/response'
+import { BasicField } from '../../types'
+
+import { FormSgSdk } from './FormSgSdkService'
+
+/**
+ * Encrypts given submission responses with the public key.
+ * @param responses the responses to encrypt
+ * @param publicKey the public key to encrypt with
+ * @returns the encrypted responses
+ * @throws error if the given responses are malformed
+ */
+export const encryptSubmissionResponses = (
+  responses: unknown,
+  publicKey: string,
+): string => {
+  return FormSgSdk.crypto.encrypt(validateResponses(responses), publicKey)
+}
+
+const isPossibleResponse = (
+  o: unknown,
+): o is Record<string, unknown> & { fieldType: BasicField } => {
+  const isPossibleObject = typeof o === 'object' && o !== null
+  if (!isPossibleObject) return false
+
+  return !!(o as Record<string, unknown>).fieldType
```

Comment:
Will update check

Reply:
Changed the fieldType check to be string instead of checking whether fieldType exists in BasicField enum (don't want to create a new array every time to check when the switch statement below is going to catch everything)

Reply:
Updated in 460274d